SpiceSharp/Algebra/Solve/ComplexSolver.cs

```diff
@@ -0,0 +1,333 @@
+                        // intermediate[row] -= temp * element
+                        intermediate[element.Row] -= temp * element.Value;
+                        element = element.Below;
+                    }
+                }
+            }
+
+            // Backward substitution
+            for (int i = Matrix.Size; i > 0; i--)
+            {
+                Complex temp = intermediate[i];
+                var pivot = Matrix.GetDiagonalElement(i);
+                var element = pivot.Right;
+
+                while (element != null)
+                {
+                    // temp -= element * intermediate[column]
+                    temp -= element.Value * intermediate[element.Column];
+                    element = element.Right;
+                }
+                intermediate[i] = temp;
+            }
+
+            // Unscrable
+            Column.Unscramble(intermediate, solution);
+        }
+
+        /// <summary>
+        /// Find the solution for a factored matrix and a right-hand-side (transposed)
+        /// </summary>
+        /// <param name="solution">Solution vector</param>
+        public override void SolveTransposed(Vector<Complex> solution)
+        {
+            if (solution == null)
+                throw new ArgumentNullException(nameof(solution));
+
+            // TODO: Maybe we should cache intermediate
+            // Scramble
+            var intermediate = new Complex[Matrix.Size + 1];
+            var rhsElement = Rhs.First;
+            while (rhsElement != null)
+            {
+                int newIndex = Column[Row.Reverse(rhsElement.Index)];
+                intermediate[newIndex] = rhsElement.Value;
+                rhsElement = rhsElement.Next;
+            }
+
+            // Forward elimination
+            for (int i = 1; i <= Matrix.Size; i++)
+            {
+                Complex temp = intermediate[i];
+
+                // This step of the elimination is skipped if temp equals 0
+                if (!temp.Equals(0.0))
+                {
+                    var element = Matrix.GetDiagonalElement(i).Right;
+                    while (element != null)
+                    {
+                        // intermediate[col] -= temp * element
+                        intermediate[element.Column] -= temp * element.Value;
+                        element = element.Right;
+                    }
+                }
+            }
+
+            // Backward substitution
+            for (int i = Matrix.Size; i > 0; i--)
+            {
+                Complex temp = intermediate[i];
+                var pivot = Matrix.GetDiagonalElement(i);
+                var element = pivot.Below;
+                while (element != null)
+                {
+                    // temp -= intermediate[element.row] * element
+                    temp -= intermediate[element.Row] * element.Value;
+                    element = element.Below;
+                }
+
+                // intermediate = temp / pivot
```

Comment:
@svenboulanger Is comment ok?

Reply:
pivot = 1 / pivot during factorization to save on division operations.